builtin/logical/pki/key_util.go

```diff
@@ -0,0 +1,126 @@
+package pki
+
+import (
+	"context"
+	"crypto"
+	"encoding/pem"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/vault/sdk/helper/certutil"
+	"github.com/hashicorp/vault/sdk/helper/errutil"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+type managedKeyContext struct {
+	ctx        context.Context
+	b          *backend
+	mountPoint string
+}
+
+func newManagedKeyContext(ctx context.Context, b *backend, mountPoint string) managedKeyContext {
+	return managedKeyContext{
+		ctx:        ctx,
+		b:          b,
+		mountPoint: mountPoint,
+	}
+}
+
+func comparePublicKey(ctx managedKeyContext, key *keyEntry, publicKey crypto.PublicKey) (bool, error) {
```

Comment:
Maybe it makes sense to add a "fast path" and compare on bytes? Are there any cases where two bytes are _equal_ but their public keys would be unequal? 

That would avoid having to (unnecessarily) parse string->PEM->ASN.1->GoStruct blobs repeatedly. Just a thought.

Reply:
Yeah as you pointed out below not sure if it's worth it at least for this round to add in a fast path.

Reply:
Nah that's fair.